lambdas/data-migration2/tests/test-executions.js

```diff
@@ -0,0 +1,456 @@
+});
+
+test.serial('migrateExecutionRecord correctly migrates execution record', async (t) => {
+  // This will be the top-level execution (no parent execution)
+  const fakeExecution = fakeExecutionFactoryV2({ parentArn: undefined });
+  const fakeCollection = fakeCollectionFactory();
+  const fakeAsyncOperation = fakeAsyncOperationFactory({
+    id: uuid(),
+    output: '{ "output": "test" }',
+  });
+
+  const [
+    existingExecution,
+    existingCollection,
+    existingAsyncOperation,
+  ] = await Promise.all([
+    executionsModel.create(fakeExecution),
+    collectionsModel.create(fakeCollection),
+    asyncOperationsModel.create(fakeAsyncOperation),
+  ]);
+
+  t.teardown(() => Promise.all([
+    executionsModel.delete({ arn: fakeExecution.arn }),
+    collectionsModel.delete(fakeCollection),
+    asyncOperationsModel.delete({ id: fakeAsyncOperation.id }),
+  ]));
+
+  // migrate existing async operation and collection
+  await migrateAsyncOperationRecord(existingAsyncOperation, t.context.knex);
+  await migrateCollectionRecord(existingCollection, t.context.knex);
+
+  // migrate the existing dynamo execution to postgres so
+  // we can use it as the parent for the next execution
+  await migrateExecutionRecord(existingExecution, t.context.knex);
+
+  const existingPostgresExecution = await t.context.knex.queryBuilder()
+    .select()
+    .table('executions')
+    .where({ arn: existingExecution.arn })
+    .first();
+
+  // Create new Dynamo execution to be migrated to postgres
+  const newExecution = fakeExecutionFactoryV2({
+    parentArn: existingExecution.arn,
+    collectionId: `${existingCollection.name}___${existingCollection.version}`,
+    asyncOperationId: existingAsyncOperation.id,
+  });
+
+  await migrateExecutionRecord(newExecution, t.context.knex);
+
+  const createdRecord = await t.context.knex.queryBuilder()
+    .select()
+    .table('executions')
+    .where({ arn: newExecution.arn })
+    .first();
+
+  assertPgExecutionMatches(t, newExecution, createdRecord, {
+    async_operation_cumulus_id: 1,
+    collection_cumulus_id: 1,
+    parent_cumulus_id: existingPostgresExecution.cumulus_id,
+  });
+});
+
+test.serial('migrateExecutionRecord throws error on invalid source data from Dynamo', async (t) => {
+  const newExecution = fakeExecutionFactoryV2();
+
+  // make source record invalid
+  delete newExecution.arn;
+
+  await t.throwsAsync(migrateExecutionRecord(newExecution, t.context.knex));
+});
+
+test.serial('migrateExecutionRecord handles nullable fields on source execution data', async (t) => {
+  const newExecution = fakeExecutionFactoryV2();
+
+  // // remove nullable fields
+  delete newExecution.asyncOperationId;
+  delete newExecution.collectionId;
+  delete newExecution.tasks;
+  delete newExecution.error;
+  delete newExecution.duration;
+  delete newExecution.originalPayload;
+  delete newExecution.finalPayload;
+  delete newExecution.timestamp;
+  delete newExecution.parentArn;
```

Comment:
I _think_ we need to add `type` and `cumulusVersion` to these options

Reply:
Awesome catch. I just added those fields to the translation and forgot to add them here.